web/wp-content/themes/bumblebee/functions.listicle.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Bumblebee functions and definitions
+ *
+ * @link https://developer.wordpress.org/themes/basics/theme-functions/
+ *
+ * @package bumblebee
+ */
+
+add_filter(
+	'the_content',
+	function( $content ) {
+		$delimiter    = '<div class="tmbi-card">';
+		$cards        = array_filter( explode( $delimiter, $content ) );
+		$total_cards  = count( $cards );
+		$current_card = 0;
+		$section_num  = 0;
+		for ( $j = 0; $j <= $total_cards - 1; $j = $j + 3 ) :
+			$section_num++;
+			?>
+			<section class="content pure-g">
```

Comment:
Markup this complex should be in template file, so frontend devs can alter it without having to touch PHP logic.
A possible alternative would be to create a `the_cards` function or similar, that returns an array with the markup of each card. Then from template one could loop through them and render each card with whatever markup we need, controlling ad cadence right from the template, etc.